src/compiler/core.ts

```diff
@@ -92,6 +92,41 @@ module ts {
         return result;
     }
 
+    export function reduce<T>(array: T[], f: (a: T, x: T) => T): T;

```

Comment:
reduceLeft, otherwise i'll always be wondering which direction it's going from when i see reduce in the code.  In functional you always have foldr and foldl to make it clear (right?)